### ISSUE TITLE
Custom jQuery builds

### ISSUE THREAD

**Issue by staabm** [jquery/learn.jquery.com#653]
I am missing a article/some document which describes how to build a custom jquery version, to reduce necessary code size.

**Comment by arthurvr** [comment]
This is a topic people are interested in so :+1: from my side.

I've always wanted to hold off on writing something like that for https://github.com/jquery/jquery/issues/1691 but that might not happen in the near future so I'm not sure.

Related: [there's a pretty awesome talk](https://www.youtube.com/watch?v=3_bqLynJpWg) from @timmywil from back in San Diego. Also see [the readme](https://github.com/jquery/jquery#modules) on this topic.

**Comment by timmywil** [comment]
I haven't heard one of my talks referred to as "awesome", but thanks! :dancer:

**Comment by staabm** [comment]
at best we would have a up 2 date version of http://projects.jga.me/jquery-builder/

if this would be a thing, I wouldn't even need docs to build it on my own ;).
the mentioned project is searching for a new maintainer, maybe the jquery foundation is interessted in it...
https://github.com/jgallen23/jquery-builder/issues/22

**Comment by dmethvin** [comment]
There's an open issue at https://github.com/jquery/jquery/issues/1691 for a download builder. The basic steps are in the core README at https://github.com/jquery/jquery#modules so it seems like additional docs at the learn site would be redundant.

**Comment by arthurvr** [comment]
> There's an open issue at jquery/jquery#1691 for a download builder. The basic steps are in the core README at https://github.com/jquery/jquery#modules so it seems like additional docs at the learn site would be redundant.

@dmethvin Well, I agree that the content would be redudant. Though, where are people searching for information like this? The learning center or the readme? How many people even look at the readme? Not sure myself.

**Comment by dmethvin** [comment]
I'm not sure either. If we put it at the learn site I would think it's more likely to get out of sync though, especially regarding details like what can be excluded. Maybe have a paragraph that says it can be done and then link to the README, eventually to the download builder.

**Comment by scottgonzalez** [comment]
> Though, where are people searching for information like this? The learning center or the readme?

That's a tough question to answer. I can only provide very vague numbers that don't take intent into account.

The main page for the GitHub repo gets:
* ~18k views per month
* ~12k users per month

The learn site gets:
* ~26k sessions per **day**
* ~1M views per month
* ~470k users per month

Only a fraction of the GitHub views are going to result in someone reading the README, and only a fraction of those users are going to care about custom builds. And of course the learn site traffic is spread across various content. It would take a fair amount of investigation to try to glean information about intent based on which pages are viewed and which search terms and referrers bring users to the site.

**Comment by staabm** [comment]
What do you think of describing the "technical" parts of modular builds (so which modules exists etc) in the readme and the actual tutorial/docs on how the steps are into the learn.jquery.com website. Those 2 resources should be X-linked then..

**Comment by arthurvr** [comment]
Maybe we should punt this till https://github.com/jquery/jquery/issues/1691 happens? If it does the whole situation regarding custom builds is gonna change (and might even make articles about it needless).

@scottgonzalez Thanks for the insights. 

@staabm That's reasonable. Maybe we should keep the technical things in the readme and have an explanation about the why (and a little of the how) on the learning center. WDYT?

**Comment by staabm** [comment]
> Maybe we should punt this till jquery/jquery#1691 happens? If it does the whole situation regarding custom builds is gonna change (and might even make articles about it needless).

Agree, see above
> at best we would have a up 2 date version of http://projects.jga.me/jquery-builder/


>@staabm That's reasonable. Maybe we should keep the technical things in the readme and have an explanation about the why (and a little of the how) on the learning center. WDYT?

Also agree with that.

**Comment by donnut** [comment]
For jQuery users that are maybe not that familiar working with node and git and want a custom build we created [jQueryConfig](http://www.jqueryconfig.com)

**Comment by staabm** [comment]
@donnut maaan this tool looks pretty awesome.

**Comment by dmethvin** [comment]
Is there a reason to email the file? I noticed a pretty lengthy privacy policy that seems to indicate you may use the email for other purposes.

**Comment by mopee** [comment]
Yes you are right, we are asking for an email address.

The idea is that in the future we might mail about our services, 
e.g. to inform that the options of jQueryConfig are extended or a 
new jQuery release is out, a request to participate in a (user) survey
among front-end-developers, etc. Of course one can opt-out.

To explain:
Mope is a techstarter developing tools for front-end developers. These
tools are in developing stage.

We like to try to test our innovative ideas, but find it difficult 
to find a user test group that is familiar with javascript but has 
no expert skills.
On meetups we don't find them.
We hope to find our audience in this way.
We offer something only for this group that is relevant and useful:
jqueryconfig
They get it for free and in return we might ask them in the future to
help us by giving critique on our ideas.

**Comment by staabm** [comment]
@mopee I agree that having the email address in this spot is a major pain point. would it be possible to pre-select a opt-in for your "newsletter" and let the user decide to de-select the option (and therefore direct download the custom build, without a need to post the email)?
